src/cbmc/bmc.h

```diff
@@ -98,6 +100,13 @@ class bmct:public safety_checkert
   virtual void show_vcc_plain(std::ostream &out);
   virtual void show_vcc_json(std::ostream &out);
 
+  trace_optionst trace_options()
+  {
+    return trace_optionst {
+      options.get_bool_option("trace-json-extended"),
+    };
+  }
```

Comment:
Copying from a previous comment: Could options parsing please be moved to a single, common location? I'd just make this a member of trace_optionst.